src/NHibernate/Tool/hbm2ddl/SchemaMetadataUpdater.cs

```diff
@@ -58,9 +58,21 @@ public static void QuoteTableAndColumns(Configuration configuration, Dialect.Dia
 			{
 				QuoteTable(cm.Table, dialect);
 			}
+
 			foreach (var cm in configuration.CollectionMappings)
 			{
 				QuoteTable(cm.Table, dialect);
+
+				if(cm.Key?.ColumnIterator != null)
+				{	
+					foreach (Column column  in cm.Key.ColumnIterator)
```

Comment:
I think it's better to extract it to `void QuoteColumns(IEnumerable<ISelectable> columns)` and reuse in `QuoteTable`. What about the code above? Shouldn't it also be done to for `ClassMapping.Key.ColumnIterator`?

Reply:
And  are you sure it's  safe 'ISelectable` -> 'Column` conversions? It seems better to call `OfType<Column>()`  here

Reply:
I copy this part from:

https://github.com/nhibernate/nhibernate-core/blob/21975f6137738633fc6d99951360606e8c2298fc/src/NHibernate/Persister/Collection/AbstractCollectionPersister.cs#L233-L238


But I believe it's better user ```OfType<Column>()```

Reply:
Method extracted to ```void QuoteColumns(IEnumerable<Column> columns)``` a reuse in ```QuoteTable``` and now I'm using with ```ClassMapping.Key.ColumnIterator```